charmhelpers/core/host.py

```diff
@@ -1074,3 +1074,14 @@ def install_ca_cert(ca_cert, name=None):
     log("Installing new CA cert at: {}".format(cert_file), level=INFO)
     write_file(cert_file, ca_cert)
     subprocess.check_call(['update-ca-certificates', '--fresh'])
+
+
+def arch():
```

Comment:
As this is an "expensive" function that's never going to change it's return value once called, it's probably useful to add `@cached` to the function:

```python
@cached
def arch():
    ...
```

`@cached` is available in charmhelpers/core/hookenv.py ; hookenv is imported at the top of the file.